### PR TITLE
Empty nodes removed (related to wallabag #3455)

### PR DESCRIPTION
**Pull request by Simounet**
Just as I told you, this is the way I found to avoid empty node being fetched. It can fix issues like [this one](https://github.com/wallabag/wallabag/issues/3455). If you have a better idea, let me know.

Have a nice day.

### CODE REVIEW

**Comment by j0k3r** [comment]
Could you rebase against the master?
And also, add a test to cover that case (maybe using the framablog link in the `GrabyFunctionalTest.php`)

**Comment by j0k3r** [comment]
You can let the call happen :)
But move it to `GrabyFunctionalTest`, you'll see a lot of real test case targeting existing website in that file.

File: tests/GrabyFunctionalTest.php [hunk 1]
```diff
@@ -583,4 +583,25 @@ public function testJsonLd()
...
+    {
+        $response = new Response(
+            200,
+            [
+                'content-type' => 'text/html',
+            ],
+            Stream::factory(file_get_contents(__DIR__ . '/fixtures/sites/framablog.test'))
+        );
+        $client = new Client();
+        $client->getEmitter()->attach(new Mock([$response]));
```
Comment (j0k3r) [inline tests/GrabyFunctionalTest.php]: You can remove the mock, it is not used and it breaks the build: https://travis-ci.org/j0k3r/graby/jobs/388131182#L490

**Comment by Simounet** [comment]
I rolled back but the mock isn't working anyway. :/

File: tests/GrabyTest.php [hunk 2]
```diff
@@ -1436,4 +1436,23 @@ public function testEncodingUtf8ForTextPlainPage()
...
     }
+
+    public function testEmptyNodesRemoved()
+    {
+        $response = new Response(
+            200,
+            [
+                'content-type' => 'text/html',
+            ],
+            Stream::factory(file_get_contents(__DIR__ . '/fixtures/sites/framablog.test'))
```
Comment (j0k3r) [inline tests/GrabyTest.php]: I don't think the `file_get_contents` is required here.

**Comment by j0k3r** [comment]
Not yet :slightly_smiling_face: 
I'm wondering if your changes doesn't break the iframe on lemonde test

**Comment by Simounet** [comment]
Same player, play again! :D

**Pull request merged**